rdb/src/db.rs

```
@@ -617,6 +648,29 @@ impl Db {
         Ok(())
     }
 
+    pub fn get_bestpath_fanout(&self) -> Result<NonZeroU8, Error> {
+        let tree = self.persistent.open_tree(SETTINGS)?;
+        let fan = match tree.get(BESTPATH_FANOUT)? {
+            // fanout was not in db
+            None => DEFAULT_BESTPATH_FANOUT,
+            Some(value) => value[0],
```

One of the properties of using 

```rust
let value: [u8; 1] = (*value).try_into().map_err(...)
```

here is that the attempt [will fail if there are too many bytes in the value](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=d31320ac48f1db235079302cb2f3227b). If we just take the first element, we could be masking an error that leads to undefined program behavior. And if for some reason there is an empty value we'll crash.